File: F/2028-find-missing-observations/main.go
package main

func missingRolls(rolls []int, mean int, n int) []int {
	sum := 0
	for _, r := range rolls {
		sum += r
	}

	diff := mean*(n+len(rolls)) - sum
	if diff < 1*n || diff > 6*n {
		return []int{}
	}

	mod := diff % n
	base := diff / n
	minus := 0
	if mod > 0 {
		base++
		minus = base*n - diff
	}

	ret := []int{}
	for i := 0; i < n-minus; i++ {
		ret = append(ret, base)
	}

	for i := 0; i < minus; i++ {
		ret = append(ret, base-1)
	}

	return ret

}

File: F/2028-find-missing-observations/main.rs
impl Solution {
  pub fn missing_rolls(rolls: Vec<i32>, mean: i32, n: i32) -> Vec<i32> {
    match mean * (rolls.len() as i32 + n) - rolls.iter().sum::<i32>() {
      remain if remain < n || remain > 6 * n => vec![],
      remain => (0..(remain % n)).fold(vec![remain / n; n as usize], |mut res, i| {
        res[i as usize] += 1;
        res
      }),
    }
  }
}
